blocks/components/editable/index.js

```diff
@@ -247,24 +324,18 @@ export default class Editable extends wp.element.Component {
 		}
 	}
 
-	componentWillUnmount() {
-		this.onChange();
-	}
-
 	componentDidUpdate( prevProps ) {
 		if ( !! this.props.focus && ! prevProps.focus ) {
-			this.focus();
+			this.maybeFocus();
 		}
 
-		// The savedContent var allows us to avoid updating the content right after an onChange call
-		if (
-			this.props.tagName === prevProps.tagName &&
-			this.props.value !== prevProps.value &&
-			this.props.value !== this.savedContent &&
-			! isEqual( this.props.value, prevProps.value ) &&
-			! isEqual( this.props.value, this.savedContent )
-		) {
-			this.updateContent();
+		if ( this.props.tagName !== prevProps.tagName ) {
+			this.setSelection( this.selection );
+		}
+
+		if ( this.props.onChange && ! isEqual( this.props.value, this.content ) ) {
```

Comment:
Can `this.content` be `prevProps.value` ? I think I'd raised this before, but not recalling why we need the instance property. Or, put another way, is there any way we can not have the instance property and instead use `value` from props as a single source of truth?

Related concern: https://github.com/facebook/react/blob/8cac523/docs/tips/10-props-in-getInitialState-as-anti-pattern.md